---
title: "Hepatic Gene Expression Analysis"
subtitle: "22160 R for bio data science"
author: Antoine Andréoletti, Olivier Gaufrès, Amy Surry, Lea Skytthe, and Trine Søgaard
format: 
  revealjs:
    logo: DTU.png
footer: "Group 20" 
transition: slide
slide-number: true
self-contained: true
---

## Introduction

::: {style="font-size: 78%;"}
**Aim**: Investigating hepatic gene expression by comparing expression levels across:

|                                                                                  |                                                                        |
|--------------------------------------|:--------------------------------:|
| \- Healthy individuals - Patients with NAFLD - Patients with cirrhosis \### Data | \![\](doc/Pictures_for_presentation/liver_pic.png){style="width:80%;"} |

-   **RNA-seq** of human liver biopsies from patients with NAFLD, patients with cirrhosis, and healthy controls under both fasting and postprandial conditions\
-   **Meta data** containing additional information about the patients
:::

------------------------------------------------------------------------

## Data processing

::: r-stack
![](Pictures_for_presentation/data_processing1.png)

![](Pictures_for_presentation/data_processing2.png){.fragment}

![](Pictures_for_presentation/data_processing3.png){.fragment}
:::

::: notes
We did some data processing in the form of cleaning and augmenting of the data, before using it for analysis.

The meta data was cleaned by removing duplicate rows and columns and by removing unnecessary columns from the data set. Then the column names were changed to make more sense.

From the clean data set we removed excluded patients and then by comparing to the count data we only kept the data for patients that were included in both data sets.
:::

------------------------------------------------------------------------

## Data processing

::: r-stack
![](Pictures_for_presentation/data_processing4.png)

![](Pictures_for_presentation/data_processing5.png){.fragment}

![](Pictures_for_presentation/data_processing6.png){.fragment}
:::

::: notes
The raw count data consists of 31 files, one file per patient. These files were merged together to form one data set including the data for all the patient.

The cleaned count data was then transformed from the long format to the wide and while doing so we created 2 different data sets, one with the count data and one with the normalised count data.

The (normalised) count data was then compared to the meta data so we only kept the data for patients that were included in both data sets. Then we removed rows that weren’t different from zero. and added two new columns with the gene IDs and gene symbols related to the transcript IDs in the data set.

Lastly the column names were changed.
:::

------------------------------------------------------------------------

## Methods

<br><br>

![Workflow Diagram](Pictures_for_presentation/R-flowdiagram.drawio.png){style="margin-top: 30px;"}

------------------------------------------------------------------------

## Results - Descriptive analysis

::: {style="font-size: 78%;"}
**Aim**: First analysis of our datasets

|                                                                                                                                             |                                           |
|:-------------------------------------------------:|:-------------------:|
| \- Distributions are fairly equal throughout our data - Sick people are older on average - No significance in disease between men and women | \![\](results/EDA.png){style="width:85%"} |
:::

------------------------------------------------------------------------

## Results - PCA/Heatmap

Some plots

------------------------------------------------------------------------

## Results - DEA
::: {style="font-size: 78%;"}
Compare gene expression levels between groups using **DESeq2**

[Output for each gene:]{.underline} Global Mean - Fold-change - Adjusted p-value

::: {style="display: flex; align-items: center; justify-content: center;"}
![](Pictures_for_presentation/DEA-fasting.png){width="500"}

![](Pictures_for_presentation/heatmap-topN-genes-fasting.png){width="400"}
:::
:::
------------------------------------------------------------------------

## Results - GSEA

::: {style="font-size: 78%;"}
**Aim**: Investigating hepatic gene set enrichment

|                                                                                                                                                                                               |                                                                                   |
|:----------------------------------------------:|:----------------------:|
| \- C11 hepatocytes is a subtype of hepatocytes (1 of 3) - Hepatocytes produces hepatokines (hormone involved in metabolic regulation) - Some therapies try to promote hepatocyte regeneration | \![\](results/GSEA/GSEA_sick_vs_healthy_patients_fasting.png){style="width:100%"} |
:::

------------------------------------------------------------------------

## Discussion / Conclusion

-   Key points
-   Count_data derived from micro-array
-   Troubles with DESeq2 package and dplyr
-   Key points
